### DISCUSSION TITLE
Use ISR in K8S readonly root file system

### DISCUSSION THREAD

**Discussion by guoard** [vercel/next.js#48766] [Help]
### Summary

How can we use ISR and revalidation feature in Next.js when the application needs to write to the .next/server folder after build, but it's not possible due to the read-only file system? Even if we mount a docker volume to this path, it won't help because after a new build, we need the content that Next writes to this path. However, using a docker volume will overwrite the contents that Next puts in that path after build. Is there any solution to this issue?

**Comment by lahans** [comment]
May be a bit late, but please also provide any solution you have found to the problem.

Without using a custom server just for manipulating the distDir configuration, what you can do is.

1. In your Dockerfile immediately after the build, rename the .next folder, to .build
2.  Make sure .next is mounted as a volume
3. As Entry point, put a custom script prior to starting node, which checks if this is the first container running (for example by checking for the presence of package.json or similar). In case package.json does not exist, copy the files over to the .next folder

Your container is now ready to write to .next on the mounted volume.

**Comment by wwwdepot** [comment]
You can use the experimental [incrementalCacheHandlerPath](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath) instead of fs. This is better fit for k8s environment

**Comment by better-salmon** [comment]
Hello @guoard, please check out https://caching-tools.github.io/next-shared-cache. It provides a library for managing custom ISR cache configurations.

**Reply by guoard** [reply]
Thanks, I'll check it out!
